pkg/infrastructure/nutanix/clusterapi/clusterapi.go

```diff
@@ -146,44 +148,111 @@ func (p Provider) Ignition(ctx context.Context, in infracapi.IgnitionInput) ([]b
 	}
 	imgReq.Metadata = imgMeta
 
+	var imgUUID string
+	timeout := 10 * time.Minute
+	// wait for successful creation of the bootstrap image object in PC
+	if err = wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Duration: time.Minute * 2,
+		Factor:   float64(1.0),
+		Steps:    5,
+		Cap:      timeout,
+	}, func(ctx context.Context) (bool, error) {
+		uuid, e1 := createImage(ctx, nutanixCl, imgReq, imgName)
+		if e1 != nil {
+			return false, e1
```

Comment:
Notice that returning an error makes the wait abort immediately and return the error. So you might want to consider which errors are retriable and `return false, nil` in those cases.

Reply:
fixed.